### PR TITLE
RBAC

### PR DESCRIPTION
**Pull request by Chr1st0ph3rTurn3r**

### CODE REVIEW

File: docs/config_RBAC.md [hunk 1]
```diff
@@ -0,0 +1,133 @@
+---
+title: Configuring Role-Based Access Control
+sidebar_label: Configuring Role-Based Access Control
+---
+
+Role-Based Access Control provides a mechanism for an Administrator to create Access Management Roles that allow specific access to 128T resources such as routers, tenants, and services. 
+
+## Configuring Role-Based Access Control
+
+The default Administrator role has access to all configuration options, and cannot be removed. From this account, all roles and access are assigned.
```
Comment (ian-h-chamberlain) [inline docs/config_RBAC.md:10]: Might be worth mentioning the config name for this role is `admin`, unless that's covered somewhere else

**Review by ian-h-chamberlain — APPROVED** [review]

File: docs/config_RBAC.md [hunk 2]
```diff
@@ -0,0 +1,133 @@
+---
+title: Configuring Role-Based Access Control
+sidebar_label: Configuring Role-Based Access Control
+---
+
+Role-Based Access Control provides a mechanism for an Administrator to create Access Management Roles that allow specific access to 128T resources such as routers, tenants, and services.
```
Comment (migolnikov) [inline docs/config_RBAC.md:6]: as routers, tenants and services along with other authority level config objects.

File: docs/config_RBAC.md [hunk 3]
```diff
@@ -0,0 +1,133 @@
...
+
+The default Administrator role has access to all configuration options, and cannot be removed. From this account, all roles and access are assigned. 
+
+- Create Resource Groups
+
+	A Resource Group identifies a set of resources. One or more Resource Groups are assigned to an Access Management Role.
+
+- Assign Resource Groups to Authority-level Resources
+
+	Resource groups can be assigned to a top-level resource (Authority level) such as a router, a tenant, or a service. The process for assigning a resource group to a service is slightly different than assigning to a router or a tenant.
```
Comment (migolnikov) [inline docs/config_RBAC.md:18]: service and service-policy

File: docs/config_RBAC.md [hunk 4]
```diff
@@ -0,0 +1,133 @@
...
+
+Role-Based Access Control provides a mechanism for an Administrator to create Access Management Roles that allow specific access to 128T resources such as routers, tenants, and services. 
+
+## Configuring Role-Based Access Control
+
+The default Administrator role has access to all configuration options, and cannot be removed. From this account, all roles and access are assigned. 
+
+- Create Resource Groups
+
+	A Resource Group identifies a set of resources. One or more Resource Groups are assigned to an Access Management Role.
```
Comment (migolnikov) [inline docs/config_RBAC.md:14]: maybe change "a set of resources' to "a set of associated resources"?

File: docs/config_RBAC.md [hunk 5]
```diff
@@ -0,0 +1,133 @@
...
+
+- Assign Resource Groups to Authority-level Resources
+
+	Resource groups can be assigned to a top-level resource (Authority level) such as a router, a tenant, or a service. The process for assigning a resource group to a service is slightly different than assigning to a router or a tenant. 
+
+- Create Access Management Roles
+
+	An Access Management Role identifies the type of user - administrator or user. Roles are assigned capabilities or actions they can perform; read, write, and provision. Provisioning allows users to perform software lifecycle management duties, such as download software, upgrade existing installations, etc.
```
Comment (migolnikov) [inline docs/config_RBAC.md:22]: we should go away from concept of user and admin. I would rewrite it as:

An Access Management Role identifies capabilities or actions a user can perform; config read, config write, and provision. Provisioning allows users to perform software lifecycle management duties, such as download software, upgrade existing installations, etc.

File: docs/config_RBAC.md [hunk 6]
```diff
@@ -0,0 +1,133 @@
...
+4. Enter a name for the Resource Group, and click SAVE.
+
+	![Resource Group Name](/img/config_RBAC_NewResGrpstep4.png)
+
+5. Repeat this for as many Resource Groups as are required. 
+
+### Assign Resource Groups to Authority-level Resources
+
+1. Under Administration, select Configuration.
+2. Select a resource; a router or a tenant.
```
Comment (migolnikov) [inline docs/config_RBAC.md:51]: not just router or tenant

File: docs/config_RBAC.md [hunk 7]
```diff
@@ -0,0 +1,133 @@
...
+4. Click ADD.
+5. Create a name for the new role. Use a name that identifies the role in some way. 
+
+	![Name the Role](/img/config_RBAC_AMRstep3.png)
+
+6. Assign the capabilities for each role. In the Capabilities panel, click ADD and select the capability from the drop down. For example, user roles may only have read access to the resource, where an admin might have read and write access. Options are config-read, config-write, provisioning. 
+
+	![Assign to a Group](/img/config_RBAC_AMRstep5.png)
+
+7. Assign the Access Management Role to a Resource Group or groups. In the Resource Groups panel, click ADD and select the Resource Group from the drop down.
```
Comment (migolnikov) [inline docs/config_RBAC.md:105]: The logic is reversed. Assign Resource Group(s) to the Access Management Role.

File: docs/config_RBAC.md [hunk 8]
```diff
@@ -0,0 +1,133 @@
...
+3. Scroll down to Access Management Roles.
+
+	![Access Management Roles](/img/config_RBAC_AMRstep2.2.png)
+
+4. Click ADD.
+5. Create a name for the new role. Use a name that identifies the role in some way. 
+
+	![Name the Role](/img/config_RBAC_AMRstep3.png)
+
+6. Assign the capabilities for each role. In the Capabilities panel, click ADD and select the capability from the drop down. For example, user roles may only have read access to the resource, where an admin might have read and write access. Options are config-read, config-write, provisioning.
```
Comment (migolnikov) [inline docs/config_RBAC.md:101]: Assign capabilities to Access Management Role

File: docs/config_RBAC.md [hunk 9]
```diff
@@ -0,0 +1,133 @@
...
+
+6. Assign the capabilities for each role. In the Capabilities panel, click ADD and select the capability from the drop down. For example, user roles may only have read access to the resource, where an admin might have read and write access. Options are config-read, config-write, provisioning. 
+
+	![Assign to a Group](/img/config_RBAC_AMRstep5.png)
+
+7. Assign the Access Management Role to a Resource Group or groups. In the Resource Groups panel, click ADD and select the Resource Group from the drop down. 
+
+	![Resource Group](/img/config_RBAC_ARGSstep7.png)
+
+8. Select the specific resources accessible by the Access Management Role. In the Resources panel, click ADD and select resources from the drop down.
```
Comment (migolnikov) [inline docs/config_RBAC.md:109]: This should be marked as Advanced. We do not know how often this will be used, so it's hard for me to say if we should just exclude this from documentation.

File: docs/config_RBAC.md [hunk 10]
```diff
@@ -0,0 +1,133 @@
...
+Repeat this process for additional roles.
+
+### Create Users and Assign Roles
+
+1. From the Administration menu, select Users.
+2. On the Users screeen, select NEW USER in the top right corner. 
+
+	![New User](/img/config_RBAC_CUARstep2.png)
+
+3. On the New Users panel, enter the following:
```
Comment (migolnikov) [inline docs/config_RBAC.md:124]: you should mention that a User can have multiple roles.

**Review by migolnikov — CHANGES_REQUESTED** [review]

File: docs/config_RBAC.md (hunk 1 above)
Comment (Chr1st0ph3rTurn3r) [reply docs/config_RBAC.md:10]: added

File: docs/config_RBAC.md (hunk 2 above)
Comment (Chr1st0ph3rTurn3r) [reply docs/config_RBAC.md:6]: ...resources such as routers, tenants, and services, as well as other Authority-level configuration objects.

File: docs/config_RBAC.md (hunk 3 above)
Comment (Chr1st0ph3rTurn3r) [reply docs/config_RBAC.md:18]: done

File: docs/config_RBAC.md (hunk 4 above)
Comment (Chr1st0ph3rTurn3r) [reply docs/config_RBAC.md:14]: I like that. Done.

File: docs/config_RBAC.md (hunk 5 above)
Comment (Chr1st0ph3rTurn3r) [reply docs/config_RBAC.md:22]: Done.

File: docs/config_RBAC.md (hunk 6 above)
Comment (Chr1st0ph3rTurn3r) [reply docs/config_RBAC.md:51]: Select an Authority-level resource; for example, a router or a tenant.

File: docs/config_RBAC.md (hunk 7 above)
Comment (Chr1st0ph3rTurn3r) [reply docs/config_RBAC.md:105]: fixed.

File: docs/config_RBAC.md (hunk 8 above)
Comment (Chr1st0ph3rTurn3r) [reply docs/config_RBAC.md:101]: done

File: docs/config_RBAC.md (hunk 9 above)
Comment (Chr1st0ph3rTurn3r) [reply docs/config_RBAC.md:109]: reworded to indicate that it is an optional setting. 
"In some cases you may choose to make specific resources accessible to an Access Management Role. This is an optional setting. In the Resources panel, click ADD and select resources from the drop down. "

File: docs/config_RBAC.md (hunk 10 above)
Comment (Chr1st0ph3rTurn3r) [reply docs/config_RBAC.md:124]: done

File: docs/config_RBAC.md [hunk 11]
```diff
@@ -0,0 +1,133 @@
+---
+title: Configuring Role-Based Access Control
+sidebar_label: Configuring Role-Based Access Control
+---
+
+Role-Based Access Control provides a mechanism for an Administrator to create Access Management Roles that allow specific access to 128T resources such as routers, tenants, and services, as well as other Authority-level configuration objects. 
+
+## Configuring Role-Based Access Control
+
+The default Administrator role (`admin`) has access to all configuration options, and cannot be removed. From this account, all roles and access are assigned.
```
Comment (migolnikov) [inline docs/config_RBAC.md:10]: It's important to note that it does not have to be done from "admin" account, but rather from an account of type "admin". So I would reword "From this account, all .." to "From this built in access role, all ..."

**Review by migolnikov — APPROVED** [review]

File: docs/config_RBAC.md (hunk 11 above)
Comment (Chr1st0ph3rTurn3r) [reply docs/config_RBAC.md:10]: change made
